sklearn/tests/test_common.py

```diff
@@ -49,26 +51,52 @@ def test_all_estimators():
     # properly
     assert_greater(len(estimators), 0)
 
-    for name, Estimator in estimators:
-        # some can just not be sensibly default constructed
-        yield check_parameters_default_constructible, name, Estimator
 
+@pytest.mark.parametrize(
+        'name, Estimator',
+        all_estimators(include_meta_estimators=True)
+)
+def test_parameters_default_constructible(name, Estimator):
 
-def test_non_meta_estimators():
-    # input validation etc for non-meta estimators
-    estimators = all_estimators()
-    for name, Estimator in estimators:
+    # some can just not be sensibly default constructed
+    check_parameters_default_constructible(name, Estimator)
+
+
+def _tested_non_meta_estimators():
+    for name, Estimator in all_estimators():
         if issubclass(Estimator, BiclusterMixin):
             continue
         if name.startswith("_"):
             continue
+        yield name, Estimator
+
+
+def _carthesian_product_checks(check_generator, estimators):
```

Comment:
Here the idea is that we generate all valid checks for all estimators which is loosely a cartesian product (without the typo), but better naming welcome..